tests/cases/compiler/inKeywordAndUnknown.ts

```diff
@@ -16,3 +16,9 @@ function f(x: {}, y: unknown) {
     }
     y;  // {}
 }
+
+
+// repro #51007
+function isHTMLTable(table: unknown): boolean {
+    return !!table && table instanceof Object && 'html' in table;

```

Comment:
I think the test here should actually test for an access of `table.html` in some capacity - e.g.

```ts
function tryGetHtmlPropFromTable(table: unknown) {
    if (!!table && table instanceof Object && 'html' in table && table.html instanceof Object) {
        return table.html;
    }
    return undefined;
}

Reply:
hm, what would it change here? the reported error was that `in` couldn't be used at all in this situation - we could add more "assertions" here than what was originally reported but I'm not sure what this particular snippet would bring to the table here. What kind of nuance am I missing? 😅

Reply:
The original test is fine; having another test that actually tries to narrow and use the newly-tested property corresponds to something closer to real-world usage.

Reply:
Ok, I've pushed out the proposed test case - please take a look if this is what you had in mind here.